packages/code-editor/src/CodeEditor/stories/Xml.tsx

```diff
@@ -0,0 +1,322 @@
+  const [xmlTree, setXmlTree] = useState<{
+    tree: Tree;
+    attributes: Attributes;
+  }>();
+  const [defaultExpandedKeys, setDefaultExpandedKeys] = useState<string[]>([]);
+
+  const editorRef = useRef<any>(null);
+
+  const buildTree = (xml: string) => {
+    const parser = new DOMParser();
+    const xmlDoc = parser.parseFromString(xml, "application/xml");
+
+    const allKeys: string[] = [];
+    const attributes: Attributes = {};
+
+    const traverse = (node: HTMLElement | ChildNode) => {
+      const structure: Tree = {};
+      const { nodeName } = node;
+      const nodeId = uniqueId();
+      const nodeKey = `${nodeName}_${nodeId}`;
+
+      allKeys.push(nodeKey);
+      structure[nodeKey] = undefined;
+
+      // Get attributes for node
+      if ((node as any).attributes) {
+        for (let i = 0; i < (node as any).attributes.length; i++) {
+          const attr = (node as any).attributes[i];
+          attributes[nodeKey] = {
+            ...attributes[nodeKey],
+            [attr.name]: attr.value,
+          };
+        }
+      }
+
+      // Get tree structure
+      for (const child of node.childNodes) {
+        if (child.nodeType === Node.ELEMENT_NODE) {
+          const childStructure = traverse(child);
+
+          const found = Object.keys(structure).find(
+            (key) => key.split("_")[0] === nodeName,
+          );
+
+          if (found && structure[found]) {
+            if (!Array.isArray(structure[found])) {
+              structure[found] = [structure[found]];
+            }
+            if (Array.isArray(structure[found]))
+              structure[found].push(childStructure);
+          } else {
+            structure[nodeKey] = childStructure;
+          }
+        } else if (
+          child.nodeType === Node.TEXT_NODE &&
+          child.nodeValue?.trim()
+        ) {
+          structure[nodeKey] = child.nodeValue.trim();
+        }
+      }
+      return structure;
+    };
+
+    return {
+      tree: traverse(xmlDoc.documentElement),
+      keys: allKeys,
+      attributes,
+    };
+  };
+
+  const handleMount: HvCodeEditorProps["onMount"] = (editor) => {
+    editorRef.current = editor;
+  };
+
+  const handleOpenSearch: HvButtonProps["onClick"] = () => {
+    if (editorRef.current) {
+      editorRef.current.getAction("actions.find").run();
+    }
+  };
+
+  const handleClickTree: HvButtonProps["onClick"] = () => {
+    const { tree, attributes, keys } = buildTree(editorValue);
+    setXmlTree({
+      tree,
+      attributes,
+    });
+    setDefaultExpandedKeys(keys);
+    setOpened(true);
+  };
+
+  return (
+    <div>
+      <Header onClickTree={handleClickTree} onClickSearch={handleOpenSearch} />
+      <HvCodeEditor
+        height={270}
+        language="xml"
+        value={editorValue}
+        onChange={(content) => setEditorValue(content ?? "")}
+        xsdSchema={xsdSchema}
+        onMount={handleMount}
+      />
+      {opened && (
```

Comment:
Doesn't `open={opened}` already handle this? This is destroying more of the React tree than necessary

Reply:
Yep, don't know why I added this here 😅 I'll update